soroban-spec/src/gen/ts/boilerplate.rs

```diff
@@ -0,0 +1,148 @@
+        &self,
+        contract_name: &str,
+        contract_id: &str,
+        rpc_url: &str,
+        network_passphrase: &str,
+        spec: &[ScSpecEntry],
+    ) -> std::io::Result<()> {
+        self.replace_placeholder_patterns(contract_name, contract_id, rpc_url, network_passphrase)?;
+        self.append_index_ts(spec)
+    }
+
+    fn replace_placeholder_patterns(
+        &self,
+        contract_name: &str,
+        contract_id: &str,
+        rpc_url: &str,
+        network_passphrase: &str,
+    ) -> std::io::Result<()> {
+        let replacement_strings = &[
+            ("INSERT_CONTRACT_NAME_HERE", contract_name),
+            (
+                "INSERT_SCREAMING_SNAKE_CASE_CONTRACT_NAME_HERE",
+                &contract_name.to_shouty_snake_case(),
+            ),
+            (
+                "INSERT_CAMEL_CASE_CONTRACT_NAME_HERE",
+                &contract_name.to_lower_camel_case(),
+            ),
+            ("INSERT_CONTRACT_ID_HERE", contract_id),
+            ("INSERT_NETWORK_PASSPHRASE_HERE", network_passphrase),
+            ("INSERT_RPC_URL_HERE", rpc_url),
+        ];
+        let root: &Path = self.as_ref();
+        [
+            "package.json",
+            "README.md",
+            "src/constants.ts",
+            "src/env.d.ts",
+            "src/index.ts",
+            "src/invoke.ts",
+        ]
+        .into_iter()
+        .map(|file_name| {
+            let file = &root.join(file_name);
+            let mut contents = fs::read_to_string(file).unwrap();
+            replacement_strings
+                .iter()
+                .for_each(|(pattern, replacement)| {
+                    contents = contents.replace(pattern, replacement);
+                });
+            fs::write(file, contents)
+        })
+        .collect::<std::io::Result<()>>()
+    }
+
+    fn append_index_ts(&self, spec: &[ScSpecEntry]) -> std::io::Result<()> {
+        fs::OpenOptions::new()
+            .append(true)
+            .open(self.0.join("src/index.ts"))?
+            .write_all(generate(spec).as_bytes())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use assert_fs::TempDir;
+
+    use super::*;
+
+    const EXAMPLE_WASM: &[u8] =
+        include_bytes!("../../../../target/wasm32-unknown-unknown/release/test_contract_data.wasm");
+
+    fn init(root: impl AsRef<Path>) -> std::io::Result<Project> {
+        let spec = crate::read::from_wasm(EXAMPLE_WASM).unwrap();
+        let p: Project = root.as_ref().to_path_buf().try_into()?;
+        p.init(
+            "contract-data-example",
+            "C…",
+            "https://rpc-futurenet.stellar.org:443/soroban/rpc",
+            "Test SDF Future Network ; October 2022",
+            &spec,
+        )
+        .unwrap();
+        Ok(p)
+    }
+
+    #[test]
+    fn test_project_dir_location() {
+        let temp_dir = TempDir::new().unwrap();
+        let _: Project = init(&temp_dir).unwrap();
+    }
+
+    #[test]
+    fn test_project_dir_location_non_temp() {
```

Comment:
Writing into a non-tmp location in a test is a little scary. Do we need this? It shouldn't matter where the writing happens, writing is writing.

Reply:
Fair enough!